plugins/inputs/dpdk/dpdk.go

```diff
@@ -104,8 +105,8 @@ func (dpdk *dpdk) Init() error {
 	dpdk.connector = newDpdkConnector(dpdk.SocketPath, dpdk.AccessTimeout)
 	initMessage, err := dpdk.connector.connect()
 	if initMessage != nil {
-		dpdk.Log.Debugf("Successfully connected to %v running as process with PID %v with len %v",
-			initMessage.Version, initMessage.Pid, initMessage.MaxOutputLen)
+		dpdk.Log.Debugf("Successfully connected to %v running as process with PID %d with len %d",
+			internal.SanitizeInput(initMessage.Version), initMessage.Pid, initMessage.MaxOutputLen)
```

Comment:
Hmmm maybe we should do it directly in the logger? This way we would cover all cases in one spot...

Reply:
I am not sure how to do it in the logger:
```
func (l *Logger) Debugf(format string, args ...interface{}) {
	log.Printf("D! ["+l.Name+"] "+format, args...)
}
```

I guess we would have to loop through the args and check if it is of type string and then run the sanitize on it, I wonder if that would cause any performance issues?

Reply:
Not sure either, but I don't think we need performance in the logging, do we?

Reply:
@sspaink, it does sound like doing this in one place would be a bit more efficient from a maintainability standpoint. Thoughts?

Reply:
I've updated the PR to move the sanitation to the loggers themselves but looks like the codeql scanner isn't smart enough to recognize the sanitation happening in the logging method, it is still reporting these errors: `This log write receives unsanitized user input`. I could ignore them of course but not sure, might defeat the purpose of the scanner. Thoughts? Maybe I am overlooking something?

Reply:
Looks like your original method of doing this on each plugin is the way to go if we want codeql to be happy. I guess I can understand the desire to be more explicit as to when data is sanitized, but it does seem repetitive.

Reply:
This makes me doubt the usefulness of this, not sure how secure it makes Telegraf if all carriage returns are removed from logs. Maybe time to abandon this tool?

Reply:
I'm inclined to agree with your last comment. I do think it might be worth running quarterly, for example to see if we have missed something obvious. Otherwise, closing seems like the next step.

Reply:
@srebhan brought up gosec, and I remembered we have an [issue](https://github.com/influxdata/telegraf-internal/issues/276) to enable it in golangci. It might be worth look into his as an alternative.

Reply:
I think it would be worth looking into gosec over this, closing this in the meantime.